dialog/form.go

```diff
@@ -0,0 +1,86 @@
+package dialog
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/container"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+// FormDialog is like the standard Dialog but with confirm and cancel buttons.
+type FormDialog struct {
+	*dialog
+	confirm *widget.Button
+	cancel  *widget.Button
+}
+
+// NewFormDialog creates and returns a dialog over the specified application using
+// the provided FormItems. The cancel button will have the dismiss text set and the confirm button will
+// use the confirm text. The response callback is called on user action after validation passes.
+// If any Validatable widget reports that validation has failed, then the confirm
+// button will be disabled. The initial state of the confirm button will reflect the initial
+// validation state of the items added to the form dialog.
+// The MinSize() of the CanvasObject passed will be used to set the size of the window.
+func NewFormDialog(title, confirm, dismiss string, items []*widget.FormItem, callback func(bool),
+	parent fyne.Window) *FormDialog {
+	var itemObjects = make([]fyne.CanvasObject, 0, len(items)*2)
```

Comment:
This makes a new slice with zero length and `len(items)*2` capacity. I think it would be a nice little performance improvement if we instead set the length to `len(items)*2`. We would then need to change each item individually. Such as something like this:
```suggestion
	var itemObjects = make([]fyne.CanvasObject, len(items)*2)
	for i := 0; i < len(items)*2-2; i +=2 {
	    itemObjects[i] = widget.NewLabel(ii.Text)
	    itemObjects[i+1] = ii.Widget
	}
```

Reply:
Just realized that the suggestion above wouldn't work. Ignore this one.

Reply:
Not by much, but more than 0. Definitely worth the extra code. :)

Reply:
Not as typed, but I can use this.
```
	var itemObjects = make([]fyne.CanvasObject, len(items)*2)
	for ii, item := range items {
		itemObjects[ii*2] = widget.NewLabel(item.Text)
		itemObjects[ii*2+1] = item.Widget
	}
```

Reply:
That's genius. Nice work :)